pkg/asset/agent/image/oc.go

```diff
@@ -0,0 +1,220 @@
+	OcDefaultRetryDelay = time.Second * 5
+)
+
+// Config is used to set up the retries for extracting the base ISO
+type Config struct {
+	MaxTries   uint
+	RetryDelay time.Duration
+}
+
+// Release is the interface to use the oc command to the get image info
+type Release interface {
+	GetBaseIso(log logrus.FieldLogger, releaseImage string, pullSecret string, mirrorConfig []mirror.RegistriesConfig, architecture string) (string, error)
+}
+
+type release struct {
+	executer executer.Executer
+	config   Config
+}
+
+// NewRelease is used to set up the executor to run oc commands
+func NewRelease(executer executer.Executer, config Config) Release {
+	return &release{executer: executer, config: config}
+}
+
+const (
+	templateGetImage             = "oc adm release info --image-for=%s --insecure=%t %s"
+	templateImageExtract         = "oc image extract --file=%s --path /:/%s --confirm %s"
+	templateImageExtractWithIcsp = "oc image extract --file=%s --path /:/%s --confirm --icsp-file=%s %s"
+)
+
+// Get the CoreOS ISO from the releaseImage
+func (r *release) GetBaseIso(log logrus.FieldLogger, releaseImage string, pullSecret string, mirrorConfig []mirror.RegistriesConfig, architecture string) (string, error) {
+
+	// Get the machine-os-images pullspec from the release and use that to get the CoreOS ISO
+	image, err := r.getImageFromRelease(log, machineOsImageName, releaseImage, pullSecret)
+	if err != nil {
+		return "", err
+	}
+
+	cacheDir, err := GetCacheDir(imageDataType)
+	if err != nil {
+		return "", err
+	}
+
+	filename := fmt.Sprintf(coreOsFileName, architecture)
+	// Check if file is already cached
+	filePath, err := GetFileFromCache(filename, cacheDir)
+	if err != nil {
+		return "", err
+	}
+	if filePath != "" {
+		// Found cached file
+		return filePath, nil
+	}
+
+	path, err := r.extractFileFromImage(log, image, filename, cacheDir, pullSecret, mirrorConfig)
+	if err != nil {
+		return "", err
+	}
+	return path, err
+}
+
+func (r *release) getImageFromRelease(log logrus.FieldLogger, imageName, releaseImage, pullSecret string) (string, error) {
+	cmd := fmt.Sprintf(templateGetImage, imageName, true, releaseImage)
+
+	log.Infof("Fetching image from OCP release (%s)", cmd)
+	image, err := execute(log, r.executer, pullSecret, cmd)
+	if err != nil {
+		return "", err
+	}
+
+	return image, nil
+}
+
+func (r *release) extractFileFromImage(log logrus.FieldLogger, image, file, cacheDir, pullSecret string, mirrorConfig []mirror.RegistriesConfig) (string, error) {
+
+	var cmd string
+	if len(mirrorConfig) > 0 {
+		log.Infof("Using mirror configuration")
+		icspFile, err := getIcspFileFromRegistriesConfig(log, mirrorConfig)
+		if err != nil {
+			return "", err
+		}
+		defer removeIcspFile(icspFile)
+		cmd = fmt.Sprintf(templateImageExtractWithIcsp, file, cacheDir, icspFile, image)
+	} else {
+		cmd = fmt.Sprintf(templateImageExtract, file, cacheDir, image)
+	}
+
+	log.Infof("extracting %s to %s, %s", file, cacheDir, cmd)
```

Comment:
It would be go to use [os/exec.LookPath](https://pkg.go.dev/os/exec@go1.17.12#LookPath) to check whether the `oc` executable exists; if it doesn't then there's no need to try, and we can report the reason.

Reply:
Added a check and output a warning